hilti/toolchain/src/compiler/optimizer/passes/dead-code-cfg.cc

```diff
@@ -136,12 +141,20 @@ struct Mutator : public optimizer::visitor::Mutator {
         else if ( data->isA<Declaration>() ) {
             if ( auto* stmt = data->parent(); stmt && stmt->isA<statement::Declaration>() )
                 node = stmt;
+
+            // Declarations should keep the RHS
+            if ( auto* local = data->tryAs<declaration::LocalVariable>(); local && local->init() )
+                replace_with = builder()->statementExpression(local->init());
         }
 
         if ( ! (node && node->hasParent()) )
             return false;
 
-        removeNode(node, msg);
+        if ( replace_with )
+            replaceNode(node, replace_with, msg);
+        else
+            removeNode(node, msg);
+
         cfg->removeNode(node);
```

Comment:
This will be inconsistent now, and we rebuild the CFG after the pass anyway. Could you remove this line, and from the arg list and the comment?